jax/_src/random.py

```diff
@@ -1491,6 +1491,98 @@ def _t(key, df, shape, dtype) -> Array:
   return n * jnp.sqrt(half_df / g)
 
 
+def chisquare(key: KeyArray,
+              df: RealArray,
+              shape: Shape = (),
+              dtype: DTypeLikeFloat = dtypes.float_) -> Array:
+  """Sample Chisquare random values with given shape and float dtype.
+
+  Args:
+    key: a PRNG key used as the random key.
+    df: a float or array of floats broadcast-compatible with ``shape``
+      representing the parameter of the distribution.
+    shape: optional, a tuple of nonnegative integers specifying the result
+      shape. Must be broadcast-compatible with ``df``. The default (None)
+      produces a result shape equal to ``df.shape``.
+    dtype: optional, a float dtype for the returned values (default float64 if
+      jax_enable_x64 is true, otherwise float32).
+
+  Returns:
+    A random array with the specified dtype and with shape given by ``shape`` if
+    ``shape`` is not None, or else by ``df.shape``.
+  """
+  key, _ = _check_prng_key(key)
+  if not dtypes.issubdtype(dtype, np.floating):
+    raise ValueError(f"dtype argument to `chisquare` must be a float "
+                    f"dtype, got {dtype}")
+  dtype = dtypes.canonicalize_dtype(dtype)
+  shape = core.canonicalize_shape(shape)
```

Comment:
This will raise an error if `shape = None`, which does not align with the description in the docstring

Reply:
Hi @jakevdp, thanks for you review! I find the shape parameter for functions in `random.py` are inconsistent? For example, `multivariate_normal` requests `Optional[Shape] = None` as default, So `multivariate_normal(key, mean, cov, shape=None)` is Ok,but `cauchy`  and `double_sided_maxwell`request `shape = ()` as default, so `cauchy(..., shape=None)` or `double_sided_maxwell(..., shape=None)` raises errors, though the docstring indicates the default value,  sometimes different behavior for a sequence of these similar functions still confuse us,  shall we make a standard for this kind of parameters? For myself, I'd like to  choose `shape: Optional[Shape] = None`

Reply:
Agreed it's inconsistent, but I lean toward defaulting to `None` because I think it makes sense that if the shape is unspecified it should respect the shape of the input parameters rather than raise an error.

Reply:
Can't agree more